File: apps/mybookkeeper/frontend/src/shared/types/lease/lease-extension-summary.ts
/**
 * Mirrors backend ``LeaseExtensionSummary``.
 *
 * Exposed on ``SignedLeaseDetail.latest_extension`` for the lease detail
 * page to gate the Undo button on the 30-day undo window.
 */
export interface LeaseExtensionSummary {
  id: string;
  created_at: string; // ISO-8601 timestamp
  starts_on: string;  // YYYY-MM-DD
  ends_on: string;    // YYYY-MM-DD
  source_attachment_id: string;
}

File: apps/mybookkeeper/backend/app/schemas/leases/lease_extension_summary.py
"""Summary of the latest non-deleted lease extension.

Exposed on ``SignedLeaseResponse.latest_extension`` so the frontend can
render the Undo button conditionally. Mirrors a ``lease_term_versions``
row created with a non-null ``source_attachment_id`` (extensions only —
the seed row is never surfaced as an "extension" to the host).
"""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict


class LeaseExtensionSummary(BaseModel):
    id: uuid.UUID
    created_at: _dt.datetime
    starts_on: _dt.date
    ends_on: _dt.date
    source_attachment_id: uuid.UUID

    model_config = ConfigDict(from_attributes=True)
